tests/Aspire.Hosting.Tests/Backchannel/AppHostBackchannelTests.cs

```diff
@@ -135,6 +134,115 @@ public async Task CanStreamResourceStates()
 
         await app.StopAsync().WaitAsync(TimeSpan.FromSeconds(60));
     }
+
+    [Fact]
+    public async Task CanGetResources()
+    {
+        using var builder = TestDistributedApplicationBuilder.CreateWithTestContainerRegistry(outputHelper);
+        builder.Configuration[KnownConfigNames.UnixSocketPath] = UnixSocketHelper.GetBackchannelSocketPath();
+
+        builder.AddResource(new TestResource("test1"))
+               .WithInitialState(new () {
+                    ResourceType = "TestResource1",
+                    State = new ("Running", null),
+                    Properties = []
+               });
+
+        builder.AddResource(new TestResource("test2"))
+               .WithInitialState(new () {
+                    ResourceType = "TestResource2", 
+                    State = new ("Starting", null),
+                    Properties = []
+               });
+
+        var backchannelReadyTaskCompletionSource = new TaskCompletionSource<BackchannelReadyEvent>();
+        builder.Eventing.Subscribe<BackchannelReadyEvent>((e, ct) => {
+            backchannelReadyTaskCompletionSource.SetResult(e);
+            return Task.CompletedTask;
+        });
+
+        using var app = builder.Build();
+
+        await app.StartAsync().WaitAsync(TimeSpan.FromSeconds(60));
```

Comment:
The `DefaultTimeout` method is better at this. It provides a better exception message, adapts its time depending on whether running locally or on the server, and doesn't timeout while running with the debugger attached.